### DISCUSSION TITLE
Tasmota 20A Device with Temp & Power Meter?

### DISCUSSION THREAD

**Discussion by manoloxxl** [arendst/Tasmota#18309] [Q&A]
Hi,

I’m wondering if there is a Tasmota able device that can measure power consumption and is able to handle a DS18B20 (or similar)?

Didn’t found one :( Also the new THR320D / Elite Serie can just handle one of both.

Recently i’m using a SonOff R2 POW (without any logic) and behind this a THR320D to switch by rules depending on sensor temp.

I would like to have it in a single device - it should handle 16A, better 20A as the switch is controlling a 3000W heater and this killed some Pow R2 during switching.

Does anyone has an idea?

Thanks!

BTW: don’t want to use the R2 POW with an Sensor as the device controls a whirlpool and r2 is not Potential free an GPIO!

**Comment by sfromis** [comment]
Better have the temperature sensor in a separate device, without mixing it up with the energy monitoring. Due to the "low voltage" side of energy monitoring devices usually being connected to mains power (yes, I mean that as written), you should not expect to connect sensors to them.

**Comment by dukeofnukem815** [comment]
I actually had a small run-in with 230V AC as I tried to be clever hooking up multiple DS18B20s to a Sonoff Elite POWR320D to achieve exactly the thing You are looking for (just a 2500W heater though).

Flashing Tasmota as usual of course without mains and then wiring the sensors to the GPIOs I had no problems installing the whole construct - for some reason the sensor data wouldn't show as it did before when the unit was only powered with 3.3V from the UART programmer. When pulling the Jumper Wires to attach a different sensor a breadboard pin gave me quite a sting.

From my pov each and every blogger, author of tutorials and so forth dealing with ESP devices that are integrated in any kind of mains circuit showing this behaviour should put up a distinct warning flashing in bold red letters pitch 20 - In the end I did plan to switch my heatpump for the pool with one of these... 0o0

**Reply by sfromis** [reply]
The problem is not so much getting mains power to the board, but specifically about energy monitoring devices, where the easiest build is to have the ground plane connected to mains. Works without problems, as long as you do not think that adding stuff is a good plan.

What you thought was "low voltage" is by design connected to mains power. And yes, that works just fine, as long as it can maintain 3.3V potential difference between internal vcc and ground. It does not matter much what their potential difference to earth is, just keeping 3.3V relative potential. Just do not do any tinkering with the innards, unless you have a strong understanding of the electrics and all the needed safety rules for dangerous voltages.

While I'm no noob around those things, I'd just not start to wire stuff to the innards of energy monitoring devices.
